src/shared/sleep-config.c

```diff
@@ -108,77 +111,192 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
-/* If battery percentage capacity is less than equal to 5% return success */
-int battery_is_low(void) {
+/* Get the list of battery */
+static int get_sd_device_battery(sd_device_enumerator **ret) {
         int r;
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
 
-         /* We have not used battery capacity_level since value is set to full
-         * or Normal in case acpi is not working properly. In case of no battery
-         * 0 will be returned and system will be suspended for 1st cycle then hibernated */
+        assert(ret);
 
-        r = read_battery_capacity_percentage();
-        if (r == -ENOENT)
-               return false;
+        r = sd_device_enumerator_new(&e);
         if (r < 0)
-               return r;
+                return log_oom();
+
+        r = sd_device_enumerator_allow_uninitialized(e);
+        if (r < 0)
+                return r;
```

Comment:
If we request the property below, then that means all enumerated devices must be initialized. Hence, this call is not necessary.

Reply:
ok. removed.